parts/linux/cloud-init/artifacts/cse_main.sh

```diff
@@ -220,6 +220,25 @@ if [ "${IPV6_DUAL_STACK_ENABLED}" == "true" ]; then
     logs_to_events "AKS.CSE.ensureDHCPv6" ensureDHCPv6
 fi
 
+# For systemd in Azure Linux, UseDomains= is by default disabled for security purposes. Enable this
+# configuration within Azure Linux AKS that operates on trusted networks to support hostname resolution
+if [[ $OS == $MARINER_OS_NAME ]]; then
+    NETWORK_CONFIG_FILE="/etc/systemd/networkd.conf"
```

Comment:
could we move the logic inside the if-statement to it's own function, maybe in cse_config?

Reply:
Have refactored the implementation and re-validated the changes.